Description: Revert commit porting to newer features that Jammy doesn't have yet
Author: Simon Quigley <tsimonq2@ubuntu.com>
Origin: vendor
Applied-Upstream: 39a03962d27eb2d786b17af517c50075cdf678d1, revert
Last-Update: 2023-11-12
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ option(SG_EXT_EDIT "Enable ability to ed
 option(UPDATE_TRANSLATIONS "Update source translation translations/*.ts files" OFF)
 
 # Minimum Versions
-set(KF5_MINIMUM_VERSION "5.101.0")
+set(KF5_MINIMUM_VERSION "5.36.0")
 set(QT_MINIMUM_VERSION "5.15.0")
 set(QTXDG_MINIMUM_VERSION "3.12.0")
 
--- a/src/core/core.cpp
+++ b/src/core/core.cpp
@@ -32,8 +32,7 @@
 #include <XdgMimeApps>
 #include <qt5xdg/XdgDesktopFile>
 
-#include <KWindowSystem/KWindowSystem>
-#include <KWindowSystem/KX11Extras>
+#include <KF5/KWindowSystem/KWindowSystem>
 #include <xcb/xfixes.h>
 
 #ifdef X11_XCB_FOUND
@@ -269,15 +268,15 @@ void Core::getActiveWindow() // called o
     if (screen == nullptr)
         screen = QGuiApplication::screens().at(0);
 
-    WId wnd = KX11Extras::activeWindow();
+    WId wnd = KWindowSystem::activeWindow();
 
     // this window screenshot will be invalid
     // if there's no active window or the active window is ours
-    bool invalid(!wnd || !KX11Extras::hasWId(wnd) || (_wnd && _wnd->winId() == wnd));
+    bool invalid(!wnd || !KWindowSystem::hasWId(wnd) || (_wnd && _wnd->winId() == wnd));
     if (!invalid)
     { // or if it isn't on the current desktop
         KWindowInfo info(wnd, NET::WMDesktop);
-        invalid = info.valid() && !info.isOnDesktop(KX11Extras::currentDesktop());
+        invalid = info.valid() && !info.isOnDesktop(KWindowSystem::currentDesktop());
         if (!invalid)
         { // or if it is a desktop or panel/dock
             info = KWindowInfo(wnd, NET::WMWindowType);
